src/test/outmdsd/testsocket.cc

```diff
@@ -292,24 +300,39 @@ MultiSenderReaderTest(
     auto sendersCV = std::make_shared<CounterCV>(nSenders);
     auto readersCV = std::make_shared<CounterCV>(nReaders);
 
+    // use std::promise to make sure all threads are ready before
+    // master thread moves ahead
+    std::vector<std::promise<void>> thReadyList;
+    thReadyList.reserve(nSenders+nReaders);
+    for (int i = 0; i < nSenders+nReaders; i++) {
+        thReadyList.push_back(std::promise<void>());
+    }
+
     std::vector<std::future<void>> senderList;
     std::vector<std::future<void>> readerList;
 
     for (int n = 0; n < nSenders; n++) {
         auto senderT = std::async(std::launch::async, DoSend, sockClient, n,
-            nmsgs, testSource, testData, sendersCV);
+            nmsgs, testSource, testData, sendersCV, std::ref(thReadyList[n]));
         senderList.push_back(std::move(senderT));
     }
 
     size_t totalClientRead = 0;
     for (int i = 0; i < nReaders; i++) {
         auto readerT = std::async(std::launch::async, DoRead, sockClient,
-            std::ref(totalClientRead), readersCV);
+            std::ref(totalClientRead), readersCV, std::ref(thReadyList[i+nSenders]));
         readerList.push_back(std::move(readerT));
     }
 
+    // wait for all threads to be ready
+    for (auto & t : thReadyList) {
+        t.get_future().wait();
+    }
+
     // wait until all senders are finished
-    BOOST_CHECK_EQUAL(true, sendersCV->wait_for(500));
+    // NOTE: the wait timeout should depend on nmsgs, and machine hardware.
+    // Allocate enough time so tests can run on slow machines.
+    BOOST_CHECK_EQUAL(true, sendersCV->wait_for(500+nmsgs*5));
```

Comment:
>BOOST_CHECK_EQUAL(true, sendersCV->wait_for(500+nmsgs*5)); [](start = 4, length = 58)

This is what BOOST_CHECK_MESSAGE is for.
auto timeout = 500 + nmsgs*5;
BOOST_CHECK_MESSAGE(sendersCV->wait_for(timeout), "Wait for sendersCV timed out (" << timeout << "ms)");